src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.SslCtx.cs

```diff
@@ -69,7 +83,107 @@ protected override bool ReleaseHandle()
         {
             Interop.Ssl.SslCtxDestroy(handle);
             SetHandle(IntPtr.Zero);
+            if (_gch.IsAllocated)
+            {
+                //Interop.Ssl.SslCtxSetData(this, (IntPtr)_gch);
+                _gch.Free();
+            }
+
+            if (_sslSessions != null)
+            {
+                lock (_sslSessions)
+                {
+                    foreach (string name in _sslSessions.Keys)
+                    {
+                        _sslSessions.Remove(name, out IntPtr session);
+                        Interop.Ssl.SessionFree(session);
+                    }
+                }
+            }
+
             return true;
         }
+
+        public void EnableSessionCache()
+        {
+            _sslSessions = new ConcurrentDictionary<string, IntPtr>();
+            _gch = GCHandle.Alloc(this);
```

Comment:
This method should assert/throw/something if it's called when _gch is already a valid handle.

Reply:
Most of these new methods probably would benefit from an assert/throw/etc if the handle is invalid (e.g. `Debug.Assert(!IsInvalid)`)